src/libraries/Common/src/System/Security/Cryptography/ManagedCompositeMLDsaComposer.cs

```diff
@@ -0,0 +1,597 @@
+            }
+
+            //  2.  Compute the Message representative M'.
+            //      As in FIPS 204, len(ctx) is encoded as a single unsigned byte.
+            //      Randomize the message representative
+            //
+            //          r = Random(32)
+            //          M' :=  Prefix || Domain || len(ctx) || ctx || r
+            //                                              || PH( M )
+
+#if NET
+            Span<byte> r = stackalloc byte[32];
+            RandomNumberGenerator.Fill(r);
+#else
+            // TODO: add polyfill for RandomNumberGenerator.Fill
+            byte[] rBytes = new byte[32];
+            new Random().NextBytes(rBytes);
+            Span<byte> r = rBytes;
+#endif
+
+            ReadOnlySpan<byte> M_prime;
+
+            using (CompositeMLDsaMessageEncoder encoder = new(Algorithm, context, r))
+            {
+                encoder.AppendData(data);
+                M_prime = encoder.GetMessageRepresentativeAndDispose();
+            }
+
+            //  3.  Separate the private key into component keys
+            //      and re-generate the ML-DSA key from seed.
+            //
+            //          (mldsaSeed, tradSK) = DeserializePrivateKey(sk)
+            //          (_, mldsaSK) = ML-DSA.KeyGen(mldsaSeed)
+
+            /* no-op */
+
+            //  4.  Generate the two component signatures independently by calculating
+            //      the signature over M' according to their algorithm specifications.
+            //
+            //          mldsaSig = ML-DSA.Sign( mldsaSK, M', ctx=Domain )
+            //          tradSig = Trad.Sign( tradSK, M' )
+
+            bool mldsaSigned = false;
+            bool tradSigned = false;
+
+            try
+            {
+                // TODO should we use a temporary buffer for the signature? Currently if ML-DSA succeeds but the traditional fails,
+                // then the ML-DSA signature will be observable briefly before we clear it.
+                _mldsa.SignData(M_prime, destination.Slice(32, Algorithm.MLDsaAlgorithm.SignatureSizeInBytes), Algorithm.DomainSeparator);
+                mldsaSigned = true;
+            }
+            catch (CryptographicException)
+            {
+            }
+
+            int tradBytesWritten = 0;
+            bool tradResult = false;
+
+            try
+            {
+                tradResult = _componentAlgorithm.TrySignData(M_prime, destination.Slice(32 + Algorithm.MLDsaAlgorithm.SignatureSizeInBytes), out tradBytesWritten);
+                tradSigned = true;
+            }
+            catch (CryptographicException)
+            {
+            }
+
+            //  5.  If either ML-DSA.Sign() or Trad.Sign() return an error, then this
+            //      process MUST return an error.
+            //
+            //          if NOT mldsaSig or NOT tradSig:
+            //              output "Signature generation error"
+
+            if (mldsaSigned is false || tradSigned is false)
```

Comment:
Pattern matching on a bool is a little odd looking. Why not just `!mldsaSigned || !tradSigned`?